app/waf/index.js

```diff
@@ -14,22 +14,36 @@
  * limitations under the License.
  */
 
+const fs = require('fs');
 const application = require('./dist');
 
 module.exports = application;
 
 if (require.main === module) {
   // Run the application
-  const config = {
+  let config = {
     rest: {
       port: +process.env.WAF_APP_PORT || 3000,
       host: process.env.WAF_APP_HOST || '0.0.0.0',
+      protocol: 'http',
       openApiSpec: {
         // useful when used with OASGraph to locate your application
         setServersFromRequest: true,
       },
     },
   };
+
+  if (process.env.ENABLE_HTTPS) {
+    config.rest['protocol'] = 'https';
+    config.rest['port'] = +process.env.WAF_APP_HTTPS_PORT || 443;
```

Comment:
Why not reuse the `WAF_APP_HTTPS_PORT`?

Reply:
for differentiating to (HTTP port configuration) WAF_APP_PORT

Reply:
I think it have no relation with protocol setting. 
User can start either https or http service over `WAF_APP_PORT`. 
When you add one more `WAF_APP_HTTPS_PORT`, user may be confused about our program's behavior when both `WAF_APP_HTTPS_PORT` and `WAF_APP_PORT` are set.